### PR TITLE
bump the SDK version to 3.2.0-beta.0

### PR DESCRIPTION
**Pull request by mahsamoosavi**
- Bumps the SDK version to 3.2.0-beta.0, which includes support for Orbit.

- Modifies the script to be more generalized by accepting the Orbit chain as a parameter instead of hardcoding, allowing for greater flexibility in usage.

- Adds the `OrbitTxHash` to the result, presenting both `arbTxHash` and `OrbitTxHash`. Enables checking of all retryables created in a single transaction.

- Displays the number of retryables found before entering the 'for' loop. This addition provides a quick overview of the number of tickets waiting to be processed by the script.

### CODE REVIEW

File: lib/find_retryables.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ const main = async () => {
 
   // Adding your Obit chain as a custom chain to the Arbitrum SDK
   try {
-    addCustomChain({ customChain: xai })
+    addCustomNetwork({ customL2Network: xai })
```
Comment (hkalodner) [inline lib/find_retryables.ts:36]: Maybe more of an SDK review question, but shouldn't this be `customL3Network`?

File: lib/networks.ts [hunk 2]
```diff
@@ -21,7 +17,63 @@ import {
     Xai = 660279,
   }
   
-  export const xai: Chain = {
+
+
+
+  export interface TokenBridge {
```
Comment (hkalodner) [inline lib/networks.ts:23]: Is there a reason we have to declare all of this stuff vs importing it?

File: lib/networks.ts [hunk 3]
```diff
@@ -35,7 +87,9 @@ import {
     explorerUrl: 'https://explorer.xai-chain.net',
     isArbitrum: true,
     isCustom: true,
+    blockTime: 10,
     name: 'Xai',
+    partnerChainIDs: [ChainId.ArbitrumOne],
     partnerChainID: ChainId.ArbitrumOne,
```
Comment (hkalodner) [inline lib/networks.ts:93]: I'm confused as to the difference between `partnerChainIDs` and `partnerChainID`

File: lib/find_retryables.ts (hunk 1 above)
Comment (spsjvc) [reply lib/find_retryables.ts:36]: This is again a temporary misnomer, `L2Network` really means `ArbitrumNetwork, so you could either do L2 or L3

File: lib/networks.ts (hunk 2 above)
Comment (spsjvc) [reply lib/networks.ts:23]: Yeah, you shouldn't really need any of these except `L1Network` and `L2Network`, which are exported from the SDK. As long as you declare your object to be one of those types, the type system will guide you to fill out all the properties.

File: lib/networks.ts (hunk 3 above)
Comment (spsjvc) [reply lib/networks.ts:93]: Again a temporary misnomer. 😞 `partnerChainID` is the parent chain id, and `partnerChainIDs` is the children chains. So in this case, `partnerChainIDs` should be empty.

File: lib/networks.ts [hunk 4]
```diff
@@ -35,7 +87,9 @@ import {
     explorerUrl: 'https://explorer.xai-chain.net',
     isArbitrum: true,
     isCustom: true,
+    blockTime: 10,
```
Comment (spsjvc) [inline lib/networks.ts:90]: This should be `0.25`. We added `blockTime` to `L2Network` too, to make it more consistent with `L1Network`.

**Comment by TucksonDev** [comment]
Thanks @mahsamoosavi ! Here's some general feedback:

- The output has all needed information now :muscle: , but you could maybe log first the parent transaction and then the list of retryables that were submitted in it, to make the output a bit clearer.
- Smol visual bug: the same information is shown multiple times if one transaction has multiple tickets (if it has 4, for example, the same information is repeated 4 times). You could either save the parent chain tx that you have already processed somewhere, or do the process in 2 times: first gather all parent chain transactions, and then iterate over them.
- process.env.PARENT_RPC_URL and process.env.ORBIT_RPC_URL could also be part of the config.json since now all the configuration from the orbit chain is coming from there. Otherwise, you'll have to set orbit chain config in 2 places: .env and lib/config.json .
- [less important, not blocking] process.env.FROM_BLOCK and process.env.TO_BLOCK could be arguments to pass to the script instead of env variables
- [less important, not blocking] It would be interesting to be able to load multiple configs in the config file, and choose one in the script, or by argument

File: lib/find_retryables.ts [hunk 5]
```diff
@@ -120,8 +152,16 @@ const main = async () => {
   await checkRetryablesOneOff()
 }
 
-// Calling main
-main()
+// Parsing command line arguments
+const options = yargs(process.argv.slice(2))
+  .options({
+    fromBlock: { type: 'number', default: 0 },
+    toBlock: { type: 'number', default: 0 },
```
Comment (TucksonDev) [inline lib/find_retryables.ts:159]: Default for "toBlock" could be the latest block (you can calculate it in "main" if toBlock == 0)

File: lib/find_retryables.ts [hunk 6]
```diff
@@ -95,23 +107,43 @@ const main = async () => {
...
+        console.log(
+          `${messages.length} retryable${
+            messages.length === 1 ? '' : 's'
+          } found, checking their status. Arbtxhash: ${process.env
+            .ARBISCAN!}${l2TxHash}`
+        )
+        console.log('************************************************')
+      }
+
       for (let msgIndex = 0; msgIndex < messages.length; msgIndex++) {
```
Comment (TucksonDev) [inline lib/find_retryables.ts:123]: Still see duplication of retryables and I think it comes from here. In this "for" loop, the script iterates through all messages that are retryables, for the transaction of every inboxDeliveredLog. The script is picking 4 inboxDeliveredLogs coming from the same transaction, but it iterates over all of them, yielding the same 4 retryables 4 times.

One alternative to fix this is that In a first for loop (for inboxDeliveredLogs), you gather all l2TxReceipt in an array, and then, outside the for loop, you traverse that new array and perform the for loop for the "messages".

File: lib/find_retryables.ts [hunk 7]
```diff
@@ -95,23 +107,43 @@ const main = async () => {
...
+          console.log(
+            `Ticket expired (!):\norbitTxHash: ${l3Chain.explorerUrl}/tx/${retryableTicketId}`
+          )
         } else if (status === L1ToL2MessageStatus.REDEEMED) {
-          console.log(`Ticket is succesfully redeemed: arbTxHash: ${l2TxHash}`)
+          console.log(
+            `Ticket is succesfully redeemed:\norbitTxHash: ${l3Chain.explorerUrl}/tx/${retryableTicketId}`
+          )
         }
       }
```
Comment (TucksonDev) [inline lib/find_retryables.ts:148]: Nitpick: adding a console.log('') here will separate the info from the different l2TxReceipts

**Review by TucksonDev — APPROVED** [review]
LGTM! 🚀 🔥

**Pull request merged**